finality-aleph/src/network/manager/mod.rs

```diff
@@ -39,13 +60,52 @@ impl<M: Multiaddress> AuthData<M> {
         self.node_id
     }
 
-    pub fn addresses(&self) -> Vec<M> {
-        self.addresses.clone()
+    pub fn address(&self) -> A {
+        self.address.clone()
     }
 }
 
+impl<M: Data, A: AddressingInformation + Into<Vec<M>>> From<AuthData<A>> for LegacyAuthData<M> {
+    fn from(auth_data: AuthData<A>) -> Self {
+        let AuthData {
+            address,
+            node_id,
+            session_id,
+        } = auth_data;
+        let addresses = address.into();
+        LegacyAuthData {
+            addresses,
+            node_id,
+            session_id,
+        }
+    }
+}
+
+impl<M: Data, A: AddressingInformation + TryFrom<Vec<M>>> TryFrom<LegacyAuthData<M>>
+    for AuthData<A>
+{
+    type Error = ();
```

Comment:
There was never software which could trigger this error anyway – it would require someone sending authentications with an empty `Vec` of multiaddresses.  So I made this into a unit error, to avoid muddling the code with unnecessary `Display` implementations, logs that would never fire, and even more complicated templating (to do this properly we should name the possibly arbitrary error in the `TryFrom` and then require some bounds on it and then preferably also wrap it, thus returning a wrapper error here), even though I had to go through a grueling internal fight beforehand.

Reply:
Not directly, they would only have to edit the source code. They have no incentive to do so and this code is temporary (will be removed with the support for v1 authentications), so I don't expect anyone to do that in the at most couple months this will be live. Also, this was technically possible before as well, plus the reaction to the error is obvious – "this is not a correct authentication".